### ISSUE TITLE
[Bug]: Redundant $$ symbols in some kind of formulas in .html

### ISSUE THREAD

**Issue by Chupakabra0** [mokeyish/obsidian-enhancing-export#8]
**How to reproduce?**

1. Create new note.
2. Paste this code
```
Test bug:

$$
\begin{align*}
	\begin{rcases}
        \lambda_{1}(a_{11}^{*}, \ldots, a_{22}^{*}) < 0 \\
        \lambda_{2}(a_{11}^{*}, \ldots, a_{22}^{*}) < 0 \\
	\end{rcases} & \Rightarrow \text{stable knot} \\
	
	\begin{rcases}
        \lambda_{1}(a_{11}^{*}, \ldots, a_{22}^{*}) > 0 \\
        \lambda_{2}(a_{11}^{*}, \ldots, a_{22}^{*}) < 0 \\
	\end{rcases} & \Rightarrow \text{saddle}
\end{align*}
$$
```
4. Export to html.

**Expected output:**
![image](https://user-images.githubusercontent.com/47476146/168435551-f179994e-aa14-460a-b6a1-86196ce2d112.png)

**Actual output:**
![image](https://user-images.githubusercontent.com/47476146/168435671-55563172-876c-4abb-a578-c2903b484479.png)

**Comment by mokeyish** [comment]
Thanks for the feedback. I'll find time to fix it.

You can also fix it by yourself, first use pandoc convert markdown to `native`, it will output the ast of the markdown content,  then use `lua` to  remove Redundant $$.

**Comment by mokeyish** [comment]
I have another website for math equation editing. [https://math.yish.org/](https://math.yish.org/) .  You can add your formula [here](https://github.com/mokeyish/Latex-Equation-Editor/blob/master/list_latexs.json) 😄

**Comment by mokeyish** [comment]
The `Redundant $$ symbols ` should be fixed with flowing steps.


## The commands

```shell
INPUT="/path/to/your/note.md"
OUTPUT="myout.html"

# generate AST tree
pandoc -f markdown --self-contained -s -o "${OUTPUT}.ast" -t native --mathjax="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-svg-full.js" "$INPUT"

# convert command
pandoc -f markdown --lua-filter="./math.lua" --self-contained --metadata title="uname" -s -o "$OUTPUT" -t html \
 --mathjax="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-svg-full.js" "$INPUT"
```

## The generated AST Tree： `myout.html.ast`
```yaml
Pandoc
  Meta { unMeta = fromList [] }
  [ Para
      [ Str "$$"
      , SoftBreak
      , RawInline
          (Format "tex")
          "\\begin{align*}\n    \\begin{rcases}\n        \\lambda_{1}(a_{11}^{*}, \\ldots, a_{22}^{*}) < 0 \\\\\n        \\lambda_{2}(a_{11}^{*}, \\ldots, a_{22}^{*}) < 0 \\\\\n    \\end{rcases} & \\Rightarrow \\text{stable knot} \\\\\n    \n    \\begin{rcases}\n        \\lambda_{1}(a_{11}^{*}, \\ldots, a_{22}^{*}) > 0 \\\\\n        \\lambda_{2}(a_{11}^{*}, \\ldots, a_{22}^{*}) < 0 \\\\\n    \\end{rcases} & \\Rightarrow \\text{saddle}\n\\end{align*}"
      , SoftBreak
      , Str "$$"
      ]
  ]

```
## fix with lua script `math.lua`
```lua

function Para(el)
    local content = {}
    local in_display_math = false
    for _, item in pairs(el.content) do
        if item.t == 'Str' and item.text == "$$" then
            in_display_math = not in_display_math
        else
            if in_display_math then
                if item.t == 'RawInline' and item.format == 'tex' then
                    local n = pandoc.Math('DisplayMath', '\n' .. item.text .. '\n')
                    table.insert(content, n)
                else
                    table.insert(content, item)
                end
            else
                table.insert(content, item)
            end
        end
    end
    el.content = content
    return el
end
```

**Comment by Chupakabra0** [comment]
> @Chupakabra0 Hi [Chupakabra0](https://github.com/Chupakabra0) This issue have been fixed in version 1.0.8. Do you have any other questions?

Hello! Thank you for instant feedback, help and fix. Have no questions, thanks for cool plugin)
